pypeit/core/extract.py

```diff
@@ -293,6 +293,32 @@ def extract_optimal(sciimg,ivar, mask, waveimg, skyimg, rn2_img, oprof, box_radi
     var_no = np.abs(skyimg - np.sqrt(2.0) * np.sqrt(rn2_img)) + rn2_img
 
     ispec, ispat = np.where(oprof > 0.0)
+
+    # Exit gracefully if we have no positive object profiles, since that means something was wrong with object fitting
+    if not np.any(oprof > 0.0):
```

Comment:
In specobjs.py the dictionaries for optimal and boxcar are declared but empty. They are only filled in here. I agree with you that this could be more elegantly coded, but I'm not sure what a NULL method would look like. Feel free to introduce one.